lib/spack/spack/util/crypto.py

```diff
@@ -25,17 +25,57 @@
 import sys
 import hashlib
 
+import llnl.util.tty as tty
+
+
 """Set of acceptable hashes that Spack will use."""
-hashes = dict((h, getattr(hashlib, h)) for h in [
+_hash_algorithms = [
     'md5',
     'sha1',
     'sha224',
     'sha256',
     'sha384',
-    'sha512'])
+    'sha512']
+
+
+hashes = dict()
+
 
 """Index for looking up hasher for a digest."""
-_size_to_hash = dict((h().digest_size, h) for h in hashes.values())
+_size_to_hash = dict()
+
+
+class DeprecatedHash(object):
+    def __init__(self, hash_alg, alert_fn, disable_security_check):
+        self.hash_alg = hash_alg
+        self.alert_fn = alert_fn
+        self.disable_security_check = disable_security_check
+
+    def __call__(self, disable_alert=False):
+        if not disable_alert:
+            self.alert_fn("Deprecation warning: {0} is not supported"
+                          " for security use".format(self.hash_alg))
```

Comment:
minor request: can we make this say "Deprecation warning: MD5 checksums will not be supported in future Spack releases."